Automattic/wp-calypso
# Generalize expandable sidebar heading components

#### Changes proposed in this Pull Request

This PR moves the expandable sidebar components out of Reader and into the general sidebar components in `client/layout`.  This is needed for the nav drawer redesign work (see PR #31742).  

To accommodate the nav drawer work and its a/b test, these components no support passing in a component via the `materialIcon` prop.  Once the a/b test is completed, we can return to just have an `icon` prop because we will no longer have to support both a gridicon and a Material icon on the same item.

This PR also makes these sidebar headings navigable via keyboard to improve accessibility.

#### Testing instructions



The Lists and Tags sidebar items in Reader should still work as they do before this PR.  The only change to their behavior is that they now have a `tabIndex` and can be opened and closed with the Enter key.
